### PR TITLE
Integrate common requests into MSAL

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: MSAL/src/MSALPublicClientApplication.m [hunk 1]
```diff
@@ -727,31 +774,37 @@ - (void)acquireTokenSilentForScopes:(NSArray<NSString *> *)scopes
...
 
-    error = nil;
-    MSALSilentRequest *request = [[MSALSilentRequest alloc] initWithParameters:params
-                                                                  forceRefresh:forceRefresh
-                                                                    tokenCache:self.tokenCache
-                                                              expirationBuffer:self.expirationBuffer
-                                                                         error:&error];
-    
-    if (!request)
+    MSIDDefaultTokenRequestProvider *tokenRequestProvider = [[MSIDDefaultTokenRequestProvider alloc] initWithOauthFactory:_oauth2Factory defaultAccessor:_tokenCache];
```
Comment (antrix1989) [inline MSAL/src/MSALPublicClientApplication.m]: _oauth2Factory & _tokenCache were created with default authority, but we are making request with authority provided in method parameters. Is it ok?

File: MSAL/src/MSALErrorConverter.m [hunk 2]
```diff
@@ -92,7 +114,11 @@ + (void)initialize
                              MSIDCorrelationIdKey : MSALCorrelationIDKey,
                              MSIDErrorDescriptionKey : MSALErrorDescriptionKey,
                              MSIDOAuthErrorKey: MSALOAuthErrorKey,
-                             MSIDOAuthSubErrorKey: MSALOAuthSubErrorKey
+                             MSIDOAuthSubErrorKey: MSALOAuthSubErrorKey,
+                             MSIDDeclinedScopesKey: MSALDeclinedScopesKey,
+                             MSIDGrantedScopesKey: MSALGrantedScopesKey,
+                             MSIDUserDisplayableIdkey: @(1),
+                             MSIDBrokerVersionKey: @(1)
```
Comment (jasoncoolmax) [inline MSAL/src/MSALErrorConverter.m]: Why these two maps to the same value?

File: MSAL/src/MSALPublicClientApplication+Internal.h [hunk 3]
```diff
@@ -36,6 +36,7 @@
 
 @property (nonatomic, nonnull) MSIDDefaultTokenCacheAccessor *tokenCache;
 
++ (nonnull NSOrderedSet *)defaultOIDCScopes;
```
Comment (jasoncoolmax) [inline MSAL/src/MSALPublicClientApplication+Internal.h]: :+1:

File: MSAL/src/MSALWebviewType.m [hunk 4]
```diff
@@ -40,3 +41,22 @@
...
+        case MSALWebviewTypeAuthenticationSession:
+            return MSIDWebviewTypeAuthenticationSession;
+        case MSALWebviewTypeSafariViewController:
+            return MSIDWebviewTypeSafariViewController;
+#endif
+        case MSALWebviewTypeWKWebView:
+            return MSIDWebviewTypeWKWebView;
+
+        default:
+            break;
```
Comment (jasoncoolmax) [inline MSAL/src/MSALWebviewType.m]: What happen if unexpected webview type is seen here? do we fail or set it to a default one?

File: MSAL/src/MSALPublicClientApplication.m (hunk 1 above)
Comment (oldalton) [reply MSAL/src/MSALPublicClientApplication.m]: Shouldn't happen, but addressed :)

File: MSAL/src/MSALErrorConverter.m (hunk 2 above)
Comment (oldalton) [reply MSAL/src/MSALErrorConverter.m]: Because I forgot to fix them :) Will fix

File: MSAL/src/MSALWebviewType.m (hunk 4 above)
Comment (oldalton) [reply MSAL/src/MSALWebviewType.m]: Fixed

**Pull request merged**